src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,491 @@
+            }) => {
+                let mut builder = S3::default();
+
+                builder
+                    .endpoint(endpoint_url)
+                    .access_key_id(access_key_id)
+                    .secret_access_key(secret_access_key)
+                    .region("auto")
+                    .bucket(bucket_name);
+
+                Operator::new(builder)?.finish()
+            }
+            StorageParams::Fs(ref root) => {
+                let root = if root.is_empty() {
+                    warn!("Argument `--storage-fs-root` is not set while `--storage-type` is set to `fs`. using `/tmp/datenlord_backend` as default.");
+                    "/tmp/datenlord_backend"
+                } else {
+                    root.as_str()
+                };
+
+                let mut builder = Fs::default();
+                builder.root(root);
+                Operator::new(builder)?.finish()
+            }
+        };
+
+        Ok(BackendWrapper {
+            operator,
+            block_size,
+        })
+    }
+}
+
+/// The backend wrapper of `openDAL` operator.
+#[derive(Debug)]
+pub struct BackendWrapper {
+    /// The inner `Operator`
+    operator: Operator,
+    /// Block size
+    block_size: usize,
+}
+
+impl BackendWrapper {
+    /// Create a new backend
+    #[must_use]
+    pub fn new(operator: Operator, block_size: usize) -> Self {
+        Self {
+            operator,
+            block_size,
+        }
+    }
+
+    /// Get file path by `ino`
+    fn get_file_path(ino: INum) -> String {
+        format!("{ino}/")
+    }
+
+    /// Get block path by `ino` and `block_id`
+    fn get_block_path(ino: INum, block_id: usize) -> String {
+        format!("{ino}/{block_id}.block")
+    }
+}
+
+#[async_trait]
+impl Storage for BackendWrapper {
+    async fn load_from_self(&self, ino: INum, block_id: usize) -> Option<Block> {
+        let mut block = Block::new_zeroed(self.block_size);
+
+        if let Err(e) = self
+            .operator
+            .reader(&Self::get_block_path(ino, block_id))
+            .await
+            .unwrap_or_else(|e| {
+                panic!("Failed to get a reader where ino={ino} and block={block_id}: {e}")
+            })
+            .read(block.make_mut_slice())
+            .await
+        {
+            if e.kind() == std::io::ErrorKind::NotFound {
+                None
+            } else {
+                panic!(
+                    "Failed to load a block from backend where ino={ino} and block={block_id}: {e}"
+                );
+            }
+        } else {
+            Some(block)
+        }
+    }
+
+    async fn load_from_backend(&self, _: INum, _: usize) -> Option<Block> {
+        // This storage has no backend.
+        None
+    }
+
+    async fn cache_block_from_backend(&self, _: INum, _: usize, _: Block) {
+        unreachable!("This storage has no backend, and has no cache.");
+    }
+
+    async fn store(&self, ino: INum, block_id: usize, block: Block) {
```

Comment:
The `store` should return a result, as it cannot guarantee a 100% success rate in storing.

Reply:
This will be done in the future, when we introduce error handling for `Storage`.